js/algoliasearch/click_conversion_analytics.js

```diff
@@ -1,20 +1,31 @@
 algoliaBundle.$(function ($) {
 	AlgoliaAnalytics.init({
-		applicationID: algoliaConfig.applicationId,
+		appId: algoliaConfig.applicationId,
 		apiKey: algoliaConfig.instant.apiKey
 	});
 
 	// "Click" in autocomplete
 	$(algoliaConfig.autocomplete.selector).each(function () {
-		$(this).on('autocomplete:selected', function (e, suggestion) {
-			trackClick(suggestion.objectID, suggestion.__position, suggestion.__queryID);
+		$(this).on('autocomplete:selected', function (e, suggestion, dataset, context) {
+
+			var sources = analyticsHelper.sources;
+			var source = sources.find(function(e) {
+				return e.name == dataset;
+			});
+
+			trackClick(source.indexName, suggestion.objectID, suggestion.__position, suggestion.__queryID);
 		});
 	});
 
 	// "Click" on instant search page
 	$(document).on('click', algoliaConfig.ccAnalytics.ISSelector, function() {
 		var $this = $(this);
-		trackClick($this.data('objectid'), $this.data('position'));
+		var lastResults = analyticsHelper.getLastResults();
+
+		// want to track results returned
+		if (lastResults) {
+			trackClick(lastResults.index, $this.data('objectid'), $this.data('position'), lastResults.queryID ? lastResults.queryID : lastResults._rawResults[0].queryID);
+		}
```

Comment:
This might not work for infinite hits. Worth checking whether it's a use case we want to cover.

Reply:
@tkrugg do HBK is only using pagination, not infinite hits I believe. @bsuravech do you know if the M1 module uses infinite hits at all?

Reply:
@tvweinstock We do offer infinite hits as apart of the configuration:  

**Algolia Search > Instant Search Results Page > Enable Infinite Scroll** 

If we want to incorporate it into the extension, we should cover infinite hits.